### PR TITLE
✨ Add page titles to all views

### LINKED ISSUE #550: Add page titles to all views
## Persona (link)

All users would benefit from this feature.

## Requirements

Have one unique and descriptive page title for each view we have, including the admin routes and regular routes.

## Motivation/Goal

Provide descriptive page title when user navigates through the app. It's convenient for them to get to certain page when checking their browser visit history. 

## Supporting Evidence (qual or quant)

Current page title a lacking information.
image

### PR DESCRIPTION
**Pull request by XuTheBunny**
## Motivation/Goal

Provide descriptive page title when user navigates through the app. It's convenient for them to get to certain page when checking their browser visit history.

## Feature or Improvement

Add page title:
- `/` KF Data Tracker - Studies
- `/login` KF Data Tracker - Login
- `/logout` KF Data Tracker - Logout
- `/callback` KF Data Tracker - Forwarding
- `/profile` KF Data Tracker - Profile
- `/tokens` KF Data Tracker - Developer tokens
- `/events` KF Data Tracker - Events
- `/cavatica-projects` KF Data Tracker - Cavatica projects
- `/study/new-study` KF Data Tracker - New study
- `/study/{kfId}/basic-info` KF Data Tracker - Study info for {study name}
- `/study/{kfId}/documents` KF Data Tracker - Study documents for {study name}
- `/study/{kfId}/cavatica` KF Data Tracker - Study projects for {study name}
- `/study/{kfId}/logs` KF Data Tracker - Study logs for {study name}

## UI changes

**Page titles**
Screen Shot 2019-11-26 at 1 14 22 PM
**Before:**
before
**After:**
image

## Browsers Tested

| Browser             | 1024px | 768px | 480px | 320px |
| ------------------- | :----: | ----: | ----: | ----: |
| Chrome (77)  | ✅ | ✅ | ✅ | ✅ |
| Firefox (69) | ✅ | ✅ | ✅ | ✅ |
| Safari (12)      |  ✅ | ✅ | ✅ | ✅ |
| Edge (18)      |   ✅ | ✅ | ✅ | ✅ |

Closes #550

### CODE REVIEW

File: src/views/CallbackView.js [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+import {Helmet} from 'react-helmet';
 import {auth} from '../state/auth';
 import {Container, Segment, Placeholder} from 'semantic-ui-react';
 
 const CallbackView = ({history}) => {
   auth.handleAuthentication(history);
   return (
     <>
+      <Helmet>
+        <meta charSet="utf-8" />
```
Comment (dankolbman) [inline src/views/CallbackView.js]: 🔥 Probably don't need this

File: src/documents/views/StudyFilesListView.js [hunk 2]
```diff
@@ -76,6 +77,15 @@ const StudyFilesListView = ({
...
     <Grid as={Segment} basic container columns={1}>
+      <Helmet>
+        {studyByKfId ? (
+          <title>
+            KF Data Tracker - Study documents for {studyByKfId.name}
+          </title>
+        ) : (
+          <title>KF Data Tracker - Study documents</title>
+        )}
+      </Helmet>
```
Comment (bdolly) [inline src/documents/views/StudyFilesListView.js]: Could _maybe_ compact this down to, but would test it first as I'm not sure if it would bork or not. Also may want to use the study id here as it's shorter than the name
```js 
<Helmet>
<title>
KF Data Tracker - Study documents `${studyByKfId ? 'for '+studyByKfId.name : null}`
</title>
</Helmet>
```

File: src/views/CallbackView.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
 import {auth} from '../state/auth';
 import {Container, Segment, Placeholder} from 'semantic-ui-react';
 
 const CallbackView = ({history}) => {
   auth.handleAuthentication(history);
   return (
     <>
+      <Helmet>
+        <title>KF Data Tracker - Forwarding</title>
+      </Helmet>
```
Comment (bdolly) [inline src/views/CallbackView.js]: Is this necessary as this view is just flashed to the user?

File: src/views/CavaticaBixView.js [hunk 4]
```diff
@@ -99,6 +100,9 @@ const CavaticaBixView = ({match, history}) => {
   if (error || projects.error || user.error)
     return (
       <Container as={Segment} basic>
+        <Helmet>
+          <title>KF Data Tracker - Study projects - Error</title>
```
Comment (bdolly) [inline src/views/CavaticaBixView.js]: maybe add the study id or study name so you know what study the error was for

File: src/views/CavaticaProjectsView.js [hunk 5]
```diff
@@ -62,6 +63,9 @@ const CavaticaProjectsView = () => {
   if (error)
     return (
       <Container as={Segment} basic>
+        <Helmet>
+          <title>KF Data Tracker - Cavatica projects - Error</title>
```
Comment (bdolly) [inline src/views/CavaticaProjectsView.js]: same as above _maybe_ add study id

File: src/views/LogsView.js [hunk 6]
```diff
@@ -22,6 +23,13 @@ const LogsView = ({match}) => {
     },
     pollInterval: 30000,
   });
+  const {loading: studyLoading, data: studyData} = useQuery(GET_STUDY_BY_ID, {
+    variables: {
+      kfId: match.params.kfId,
+    },
+    fetchPolicy: 'network-only',
+  });
+  const studyByKfId = studyData && studyData.studyByKfId;
```
Comment (bdolly) [inline src/views/LogsView.js]: should this be in a separate PR ?

File: src/views/StudyListView.js [hunk 7]
```diff
@@ -26,6 +27,9 @@ const StudyListView = () => {
   if (!loading && studyList.length === 0)
     return (
       <Container as={Segment} basic>
+        <Helmet>
+          <title>KF Data Tracker - Studies</title>
```
Comment (bdolly) [inline src/views/StudyListView.js]: change to `My Studies` so we match the verbiage on the page

File: src/views/CavaticaProjectsView.js (hunk 5 above)
Comment (XuTheBunny) [reply src/views/CavaticaProjectsView.js]: This is the full Cavatica projects list for ADMIN role, I will add study ID to all others under a study.

File: src/views/LogsView.js (hunk 6 above)
Comment (XuTheBunny) [reply src/views/LogsView.js]: This was added because we want to get the study name or study id to show on the title.

File: src/views/CallbackView.js (hunk 3 above)
Comment (XuTheBunny) [reply src/views/CallbackView.js]: This would be helpful in debugging if users ever stuck at the this step or it takes too long.

File: src/documents/components/FileDetail/__tests__/EditExistingFile.test.js [hunk 8]
```diff
@@ -139,4 +139,4 @@ it('edits an existing file correctly', async () => {
   });
   await wait();
   expect(tree.container).toMatchSnapshot();
-}, 8000);
+}, 10000);
```
Comment (bdolly) [inline src/documents/components/FileDetail/__tests__/EditExistingFile.test.js]: if you run your local test using `yarn test -u -i` it runs test in a series and helps prevent some of the random timeout errors. This way we don't have to just randomly adjusts these timeout values

File: src/views/__test__/NewStudyView.test.js [hunk 9]
```diff
@@ -138,4 +138,4 @@ it('renders new study view correctly --  with error on submit', async () => {
   await wait(3000);
 
   expect(tree.container).toMatchSnapshot();
-});
+}, 8000);
```
Comment (bdolly) [inline src/views/__test__/NewStudyView.test.js]: same comment as above about timeouts
